lib/deep_cover/analyser/ruby25_like_branch.rb

```diff
@@ -0,0 +1,214 @@
+# frozen_string_literal: true
+
+require_relative 'subset'
+
+module DeepCover
+  class Analyser::Ruby25LikeBranch < Analyser
+    def self.human_name
+      'Ruby25 branches'
+    end
+    include Analyser::Subset
+    SUBSET_CLASSES = [Node::Case, Node::Csend, Node::If, Node::ShortCircuit,
+                      Node::Until, Node::UntilPost, Node::While, Node::WhilePost,
+                     ].freeze
+
+    def initialize(*args)
+      super
+      @loc_index = 0
+    end
+
+    def results
+      extractor = NodeCoverageExtrator.new
+      each_node.map do |node|
+        extractor.branch_coverage(node)
+      end.to_h
+    end
+
+    class NodeCoverageExtrator < SimpleDelegator
+      def initialize(node = nil)
+        self.node = node
+        @loc_index = 0
+      end
+
+      def node=(value)
+        @delegate_sd_obj = value
+      end
+
+      def node
```

Comment:
So I think the actual reason you have a `node` method is that by default `SimpleDelegator` doesn't delegate `is_a?`. Otherwise, there is no other use for it I think.
The problem with it is that (1) it's ugly that you have to do `node.is_a?(Node)`, (2) you have `extend_elsif_range` that has a local var `node` that shadows the method even though it sometime has a different value. That's very bad style imo (even for a protected method). Suprised there isn't a rubocop about this.
I think it's debatable that `is_a?` isn't delegated, or at least redefined to return `true` for the delegated object. We can do it easily, we simple have to call `undef :is_a?`, and I think that would make things much cleaner.

Reply:
I don't consider the `Extractor` to act as a `Node`. I consider it to be an object made specifically to work on `Nodes`, and so the default target in its methods is better being a node.

Whenever I need to pass the node to a method, I prefer to pass `node` instead of `self` as it makes things clearer. I basically won't write `self` except to fetch the actual node.

If I wanted to handle `is_a?`, i wouldn't use `undef`. I would just create a method. Probably something like `#working_on?(...)`. However:

* there is a grand total of 3 `node.is_a?(...)`. Every other one is on a local variable
* `node.is_a?(...)` is widely known and very clear, an alias just feels messy.

So I find little gains from doing any change about this.

You're right for `extend_elsif_range` using node. I just didn't find a name I wanted. `possible_elsif_node` felt long and ugly. Suggestion?